src/scripts/Player.ts

```diff
@@ -3,22 +3,27 @@
  * All player variables need to be saved.
  */
 class Player {
+
+
     private static _money: KnockoutObservable<number> = ko.observable(0);
-    private static _dungeonTokens: number;
+    private static _dungeonTokens: number = 0;
     private static _caughtPokemonList = [];
-    private static _route: number;
-    private static _region: GameConstants.Region;
-    private static _pokeballs: number[] = [3, 2, 4, 1];
+    private static _route: number = 1;
+    private static _routeKills: number[] = Array.apply(null, Array(GameConstants.AMOUNT_OF_ROUTES)).map(Number.prototype.valueOf, 0);
+    private static _routeKillsNeeded: number = 10;
+    private static _region: GameConstants.Region = GameConstants.Region.kanto;
+    private static _gymBadges: GameConstants.Badge[] = [GameConstants.Badge.Boulder];
```

Comment:
It's easier and faster, but with an array of booleans we can't catch the undefined badge.
`Player.hasBadge(badge)` works correctly now, and can be used.